### ISSUE TITLE
Add comment to setScale regarding double write to member variable

### ISSUE THREAD

**Issue by timvandermeij** [mozilla/pdf.js#6158]
https://github.com/mozilla/pdf.js/blob/master/web/viewer.js#L1180 is a bit confusing as we write twice to the same member variable `this.updateScaleControls`. It is however valid code as an event handler relies on it, but we need to add a comment about that there to avoid any confusion.

**Comment by Snuffleupagus** [comment]
After some more testing of the PR #6163, I seemed to have introduced some edge-case where updating the zoom `<select>` DOM element didn't work entirely correct (e.g. when setting the zoom level to `page-width`, and the closing and restoring the tab).

Apparently the current code has some dependencies on the actual event order, but I think that should be fixable by also re-factoring the `scalechange`/`resize` event handlers.
However, I've decided to do this re-factoring in a couple of separate PRs instead, in order to make both writing and testing easier.
